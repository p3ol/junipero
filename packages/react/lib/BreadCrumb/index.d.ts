import { ReactNode, MutableRefObject, ComponentPropsWithRef } from 'react';

export declare type BreadCrumbRef = {
  isJunipero: boolean;
  innerRef: MutableRefObject<any>;
};

declare interface BreadCrumbProps extends ComponentPropsWithRef<any> {
  children?: ReactNode | JSX.Element;
  className?: string;
  maxItems?: number;
  filterItem?(children: ReactNode | JSX.Element): boolean;
  ref?: MutableRefObject<BreadCrumbRef | undefined>;
}

declare function BreadCrumb(props: BreadCrumbProps): ReactNode | JSX.Element;

export default BreadCrumb;
