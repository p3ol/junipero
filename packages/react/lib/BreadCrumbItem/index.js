import { forwardRef, useImperativeHandle, useRef } from 'react';
import { classNames } from '@junipero/core';
import PropTypes from 'prop-types';

const BreadCrumbItem = forwardRef(({
  animate,
  className,
  tag: Tag = 'span',
  ...rest
}, ref) => {
  const innerRef = useRef();

  useImperativeHandle(ref, () => ({
    innerRef,
    isJunipero: true,
  }));

  const rendered = (
    <Tag
      { ...rest }
      className={classNames(
        'item',
        className
      )}
      ref={innerRef}
    />
  );

  return animate ? animate(rendered) : rendered;
});

BreadCrumbItem.displayName = 'BreadCrumbItem';
BreadCrumbItem.propTypes = {
  tag: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.object,
    PropTypes.func,
  ]),
  animate: PropTypes.func,
};

export default BreadCrumbItem;
