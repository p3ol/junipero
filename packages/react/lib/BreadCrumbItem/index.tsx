import { ComponentPropsWithRef, ElementType, MutableRefObject, ReactNode, forwardRef, useImperativeHandle, useRef } from 'react';
import { classNames } from '@junipero/core';
import PropTypes from 'prop-types';

import { ForwardedProps } from '../utils';

export declare type BreadCrumbItemRef = {
  isJunipero: boolean;
  innerRef: MutableRefObject<any>;
};

declare interface BreadCrumbItemProps extends ComponentPropsWithRef<any> {
  className?: string;
  children?: ReactNode | JSX.Element;
  tag?: string | ElementType;
  animate?(item: ReactNode | JSX.Element): ReactNode | JSX.Element;
  ref?: MutableRefObject<BreadCrumbItemRef | undefined>;
}

const BreadCrumbItem = forwardRef(({
  animate,
  className,
  tag: Tag = 'span',
  ...rest
}: BreadCrumbItemProps, ref) => {
  const innerRef = useRef();

  useImperativeHandle(ref, () => ({
    innerRef,
    isJunipero: true,
  }));

  const rendered = (
    <Tag
      { ...rest }
      className={classNames(
        'item',
        className
      )}
      ref={innerRef}
    />
  );

  return animate ? animate(rendered) : rendered;
}) as ForwardedProps<BreadCrumbItemProps, BreadCrumbItemRef>;

BreadCrumbItem.displayName = 'BreadCrumbItem';
BreadCrumbItem.propTypes = {
  tag: PropTypes.any, // TODO fixme
  animate: PropTypes.func,
};
export default BreadCrumbItem;
