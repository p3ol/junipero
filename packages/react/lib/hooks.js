import { useContext } from 'react';

import { FieldControlContext } from './contexts';

export const useFieldControl = () => useContext(FieldControlContext);
