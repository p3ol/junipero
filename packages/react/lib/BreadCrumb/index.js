import {
  Children,
  forwardRef,
  Fragment,
  useImperativeHandle,
  useReducer,
  useRef,
} from 'react';
import { exists, classNames, mockState } from '@junipero/core';
import PropTypes from 'prop-types';

import BreadCrumbItem from '../BreadCrumbItem';

const BreadCrumb = forwardRef(({
  className,
  children,
  maxItems,
  filterItem = child => child.type === BreadCrumbItem,
  ...rest
}, ref) => {
  const innerRef = useRef();
  const [state, dispatch] = useReducer(mockState, {
    opened: false,
  });

  useImperativeHandle(ref, () => ({
    innerRef,
    isJunipero: true,
  }));

  const open = e => {
    e.preventDefault();

    dispatch({ opened: true });
  };

  const items = Children
    .toArray(children)
    .map(child => child.type === Fragment ? child.props.children : child)
    .flat()
    .filter(filterItem);
  const before = items.slice(0, Math.ceil(maxItems / 2));
  const after = items.slice(-Math.floor(maxItems / 2));

  return (
    <div
      { ...rest }
      ref={innerRef}
      className={classNames(
        'junipero',
        'breadcrumb',
        state.opened ? 'opened' : exists(maxItems) ? 'collapsed' : '',
        className,
      )}
    >
      { !exists(maxItems) || maxItems > items.length || state.opened ? items : (
        <>
          { before }
          <BreadCrumbItem tag="a" onClick={open}>...</BreadCrumbItem>
          { after }
        </>
      ) }
    </div>
  );
});

BreadCrumb.displayName = 'BreadCrumb';
BreadCrumb.propTypes = {
  maxItems: PropTypes.number,
  filterItem: PropTypes.func,
};

export default BreadCrumb;
