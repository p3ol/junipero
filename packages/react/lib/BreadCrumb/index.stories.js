import { useState } from 'react';

import BreadCrumb from './index';
import BreadCrumbItem from '../BreadCrumbItem';

export default { title: 'react/BreadCrumb' };

export const basic = () => (
  <BreadCrumb>
    <BreadCrumbItem>One</BreadCrumbItem>
    <BreadCrumbItem>Two</BreadCrumbItem>
    <BreadCrumbItem>Three</BreadCrumbItem>
  </BreadCrumb>
);

export const collapsed = () => (
  <BreadCrumb maxItems={2}>
    <BreadCrumbItem>One</BreadCrumbItem>
    <BreadCrumbItem>Two</BreadCrumbItem>
    <BreadCrumbItem>Three</BreadCrumbItem>
  </BreadCrumb>
);

export const withLinks = () => (
  <BreadCrumb>
    <BreadCrumbItem tag="a" href="#">One</BreadCrumbItem>
    <BreadCrumbItem tag="a" href="#">Two</BreadCrumbItem>
    <BreadCrumbItem>Three</BreadCrumbItem>
  </BreadCrumb>
);

export const withConditionalItems = () => {
  const [enabled, setEnabled] = useState(false);

  return (
    <>
      <p><button onClick={() => setEnabled(v => !v)}>Toggle</button></p>
      <BreadCrumb maxItems={3}>
        <BreadCrumbItem>One</BreadCrumbItem>
        <BreadCrumbItem>Two</BreadCrumbItem>
        { enabled && (
          <>
            <BreadCrumbItem>Three</BreadCrumbItem>
            <BreadCrumbItem>Four</BreadCrumbItem>
          </>
        ) }
      </BreadCrumb>
    </>
  );
}
