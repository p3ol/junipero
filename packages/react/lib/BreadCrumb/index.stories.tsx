import { useState } from 'react';

import BreadCrumb from './index';
import BreadCrumbItem from '../BreadCrumbItem';

export default { title: 'react/BreadCrumb' };

export const Basic = () => (
  <BreadCrumb>
    <BreadCrumbItem>One</BreadCrumbItem>
    <BreadCrumbItem>Two</BreadCrumbItem>
    <BreadCrumbItem>Three</BreadCrumbItem>
  </BreadCrumb>
);

export const Collapsed = () => (
  <BreadCrumb maxItems={2}>
    <BreadCrumbItem>One</BreadCrumbItem>
    <BreadCrumbItem>Two</BreadCrumbItem>
    <BreadCrumbItem>Three</BreadCrumbItem>
  </BreadCrumb>
);

export const WithLinks = () => (
  <BreadCrumb>
    <BreadCrumbItem tag="a" href="#">One</BreadCrumbItem>
    <BreadCrumbItem tag="a" href="#">Two</BreadCrumbItem>
    <BreadCrumbItem>Three</BreadCrumbItem>
  </BreadCrumb>
);

export const WithConditionalItems = () => {
  const [enabled, setEnabled] = useState(false);

  return (
    <>
      <p><button onClick={() => setEnabled(v => !v)}>Toggle</button></p>
      <BreadCrumb maxItems={3}>
        <BreadCrumbItem>One</BreadCrumbItem>
        <BreadCrumbItem>Two</BreadCrumbItem>
        { enabled && (
          <>
            <BreadCrumbItem>Three</BreadCrumbItem>
            <BreadCrumbItem>Four</BreadCrumbItem>
          </>
        ) }
      </BreadCrumb>
    </>
  );
};

export const WithItemsProp = () => (
  <BreadCrumb items={['One', 'Two', 'Three']} />
);

export const WithItemsPropCollapsed = () => (
  <BreadCrumb items={['One', 'Two', 'Three']} maxItems={2} />
);
