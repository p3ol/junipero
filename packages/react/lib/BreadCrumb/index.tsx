import {
  type RefObject,
  type ComponentPropsWithoutRef,
  type ReactNode,
  type ReactElement,
  type MouseEvent,
  Children,
  Fragment,
  useImperativeHandle,
  useRef,
  useMemo,
  useState,
} from 'react';
import { exists, classNames } from '@junipero/core';

import type { JuniperoRef, SpecialComponentPropsWithRef } from '../types';
import BreadCrumbItem from '../BreadCrumbItem';

export declare interface BreadCrumbRef extends JuniperoRef {
  items: ReactNode[];
  innerRef: RefObject<HTMLDivElement>;
}

export declare interface BreadCrumbProps
  extends SpecialComponentPropsWithRef<'div', BreadCrumbRef> {
  items?: ReactNode[];
  maxItems?: number;
  filterItem?(children: ReactNode): boolean;
}

const BreadCrumb = ({
  ref,
  className,
  children,
  items,
  maxItems,
  filterItem = child => (child as ReactElement).type === BreadCrumbItem,
  ...rest
}: BreadCrumbProps) => {
  const innerRef = useRef<HTMLDivElement>(null);
  const [opened, setOpened] = useState<boolean>(false);

  useImperativeHandle(ref, () => ({
    items,
    innerRef,
    isJunipero: true,
  }));

  const open = (e: MouseEvent<HTMLElement>) => {
    e.preventDefault();
    setOpened(true);
  };

  const availableItems = useMemo<ReactNode[]>(() => (
    items
      ? items.map((item, i) => (
        <BreadCrumbItem key={i}>{ item }</BreadCrumbItem>
      ))
      : Children
        .toArray(children)
        .map((
          child: ReactElement<ComponentPropsWithoutRef<any>>
        ) => child.type === Fragment ? child.props.children : child)
        .flat()
        .filter(filterItem)
  ), [children, items, filterItem]);

  const before = useMemo<ReactNode[]>(() => (
    availableItems.slice(0, Math.ceil(maxItems / 2))
  ), [availableItems, maxItems]);

  const after = useMemo<ReactNode[]>(() => (
    availableItems.slice(-Math.floor(maxItems / 2))
  ), [availableItems, maxItems]);

  return (
    <div
      { ...rest }
      ref={innerRef}
      className={classNames(
        'junipero breadcrumb',
        opened ? 'opened' : exists(maxItems) ? 'collapsed' : '',
        className,
      )}
    >
      { !exists(maxItems) || maxItems > availableItems.length || opened
        ? availableItems
        : (
          <>
            { before }
            <BreadCrumbItem tag="a" onClick={open}>...</BreadCrumbItem>
            { after }
          </>
        ) }
    </div>
  );
};

BreadCrumb.displayName = 'BreadCrumb';

export default BreadCrumb;
