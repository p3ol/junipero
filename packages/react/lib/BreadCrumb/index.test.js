import { fireEvent, render } from '@testing-library/react';

import BreadCrumb from '.';
import BreadCrumbItem from '../BreadCrumbItem';

describe('<BreadCrumb />', () => {
  it('should render', () => {
    const { container, unmount } = render(<BreadCrumb />);
    expect(container).toMatchSnapshot();
    unmount();
  });

  it('should render a list of BreadCrumb items', () => {
    const { container, unmount } = render(
      <BreadCrumb>
        <BreadCrumbItem>test</BreadCrumbItem>
        <BreadCrumbItem>secondTest</BreadCrumbItem>
      </BreadCrumb>
    );

    expect(container).toMatchSnapshot();

    unmount();
  });

  it('should render a collapsed BreadCrumb', () => {
    const { container, getByText, unmount } = render(
      <BreadCrumb maxItems={3}>
        <BreadCrumbItem>One</BreadCrumbItem>
        <BreadCrumbItem>Two</BreadCrumbItem>
        <BreadCrumbItem>Three</BreadCrumbItem>
        <BreadCrumbItem>Four</BreadCrumbItem>
      </BreadCrumb>
    );

    expect(container).toMatchSnapshot();

    fireEvent.click(getByText('...'));

    expect(container).toMatchSnapshot();

    unmount();
  });
});
