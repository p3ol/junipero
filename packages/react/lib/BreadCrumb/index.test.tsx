import { fireEvent, render } from '@testing-library/react';

import BreadCrumbItem from '../BreadCrumbItem';
import BreadCrumb from '.';

describe('<BreadCrumb />', () => {
  it('should render', () => {
    const { container, unmount } = render(<BreadCrumb />);
    expect(container).toMatchSnapshot();
    unmount();
  });

  it('should render a list of BreadCrumb items', () => {
    const { container, unmount } = render(
      <BreadCrumb>
        <BreadCrumbItem>test</BreadCrumbItem>
        <BreadCrumbItem>secondTest</BreadCrumbItem>
      </BreadCrumb>
    );

    expect(container).toMatchSnapshot();

    unmount();
  });

  it('should render a collapsed BreadCrumb', () => {
    const { container, getByText, unmount } = render(
      <BreadCrumb maxItems={3}>
        <BreadCrumbItem>One</BreadCrumbItem>
        <BreadCrumbItem>Two</BreadCrumbItem>
        <BreadCrumbItem>Three</BreadCrumbItem>
        <BreadCrumbItem>Four</BreadCrumbItem>
      </BreadCrumb>
    );

    expect(container).toMatchSnapshot();

    fireEvent.click(getByText('...'));

    expect(container).toMatchSnapshot();

    unmount();
  });

  it('should allow to conditionnaly add items inside fragments', () => {
    const { container, getByText, unmount } = render(
      <BreadCrumb maxItems={3}>
        <BreadCrumbItem>One</BreadCrumbItem>
        <BreadCrumbItem>Two</BreadCrumbItem>
        <>
          <BreadCrumbItem>Three</BreadCrumbItem>
          <BreadCrumbItem>Four</BreadCrumbItem>
        </>
      </BreadCrumb>
    );

    expect(container).toMatchSnapshot();

    fireEvent.click(getByText('...'));

    expect(container).toMatchSnapshot();

    unmount();
  });

  it('should allow to render from items prop', () => {
    const { container, unmount } = render(
      <BreadCrumb items={['One', 'Two', 'Three']} />
    );

    expect(container).toMatchSnapshot();

    unmount();
  });

  it('should allow to render from items prop with maxItems', () => {
    const { container, getByText, unmount } = render(
      <BreadCrumb items={['One', 'Two', 'Three', 'Four']} maxItems={3} />
    );

    expect(container).toMatchSnapshot('Collapsed');

    fireEvent.click(getByText('...'));

    expect(container).toMatchSnapshot('Opened');

    unmount();
  });
});
