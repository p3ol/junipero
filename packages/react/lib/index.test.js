import { useReducer } from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { mockState } from '@junipero/core';
import { useTimeout } from '@junipero/hooks';

import TextField from './TextField';

describe('mockState: inside react', () => {
  it('should fail update the state correctly and return the old ' +
    'state', async () => {
    const Comp = () => {
      const [state, dispatch] = useReducer(mockState, {
        value: {
          text: 'This is a test',
          content: [],
        },
      });

      useTimeout(() => {
        dispatch({ value: { ...state.value, content: ['foo'] } });
      }, 1, []);

      const onTextChange = ({ value }) => {
        dispatch({ value: { ...state.value, text: value } });
      };

      const onContentChange = () => {
        dispatch({
          value: { ...state.value, content: [...state.value.content, 'bar'] },
        });
      };

      return (
        <div>
          <TextField
            data-testid="Field"
            value={state.value.text}
            onChange={onTextChange}
          />
          <button onClick={onContentChange}>Add content</button>
          <div>Text: { state.value.text }</div>
          <div>Content: { state.value.content.join(', ') }</div>
        </div>
      );
    };

    const { container, unmount } = render(<Comp />);

    fireEvent.change(screen.getByTestId('Field'),
      { target: { value: 'This is a test 2' } });

    await waitFor(() => screen.getByText('Content: foo'));
    expect(container).toMatchSnapshot('With updated text field value');

    fireEvent.click(screen.getByText('Add content'));

    expect(container)
      .toMatchSnapshot('With updated content but with old text value');

    unmount();
  });

  it('should not fail update the state correctly and return the new ' +
    'state when using mockState with a callback', async () => {
    const Comp = () => {
      const [state, dispatch] = useReducer(mockState, {
        value: {
          text: 'This is a test',
          content: [],
        },
      });

      useTimeout(() => {
        dispatch(s => ({ value: { ...s.value, content: ['foo'] } }));
      }, 1, []);

      const onTextChange = ({ value }) => {
        dispatch(s => ({ value: { ...s.value, text: value } }));
      };

      const onContentChange = () => {
        dispatch(s => ({
          value: { ...s.value, content: [...s.value.content, 'bar'] },
        }));
      };

      return (
        <div>
          <TextField
            data-testid="Field"
            value={state.value.text}
            onChange={onTextChange}
          />
          <button onClick={onContentChange}>Add content</button>
          <div>Text: { state.value.text }</div>
          <div>Content: { state.value.content.join(', ') }</div>
        </div>
      );
    };

    const { container, unmount } = render(<Comp />);

    fireEvent.change(screen.getByTestId('Field'),
      { target: { value: 'This is a test 2' } });

    await waitFor(() => screen.getByText('Content: foo'));
    expect(container).toMatchSnapshot('With updated text field value');

    fireEvent.click(screen.getByText('Add content'));

    expect(container)
      .toMatchSnapshot('With updated content AND updated text field valud');

    unmount();
  });
});
