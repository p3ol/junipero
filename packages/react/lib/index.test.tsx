import { useReducer } from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { mockState } from '@junipero/core';
import { useTimeout } from '@junipero/hooks';

import type { FieldContent } from './types';
import TextField from './TextField';

interface State {
  value: {
    text: string;
    content: string[];
  };
}

describe('mockState: inside react', () => {
  it('should correctly update the state and return the new ' +
    'state when using mockState with a callback', async () => {
    const Comp = () => {
      const [state, dispatch] = useReducer(mockState<State>, {
        value: {
          text: 'This is a test',
          content: [],
        },
      });

      useTimeout(() => {
        dispatch(s => ({ value: { ...s.value, content: ['foo'] } }));
      }, 1, []);

      const onTextChange = ({ value }: FieldContent) => {
        dispatch(s => ({ value: { ...s.value, text: value } }));
      };

      const onContentChange = () => {
        dispatch(s => ({
          value: { ...s.value, content: [...s.value.content, 'bar'] },
        }));
      };

      return (
        <div>
          <TextField
            data-testid="Field"
            value={state.value.text}
            onChange={onTextChange}
          />
          <button onClick={onContentChange}>Add content</button>
          <div>Text: { state.value.text }</div>
          <div>Content: { state.value.content.join(', ') }</div>
        </div>
      );
    };

    const { container, unmount } = render(<Comp />);

    fireEvent.change(screen.getByTestId('Field'),
      { target: { value: 'This is a test 2' } });

    await waitFor(() => screen.getByText('Content: foo'));
    expect(container).toMatchSnapshot('With updated text field value');

    fireEvent.click(screen.getByText('Add content'));

    expect(container)
      .toMatchSnapshot('With updated content AND updated text field valud');

    unmount();
  });
});
