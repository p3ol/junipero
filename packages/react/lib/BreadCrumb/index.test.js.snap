// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreadCrumb /> should allow to conditionnaly add items inside fragments 1`] = `
<div>
  <div
    class="junipero breadcrumb collapsed"
  >
    <span
      class="item"
    >
      One
    </span>
    <span
      class="item"
    >
      Two
    </span>
    <a
      class="item"
    >
      ...
    </a>
    <span
      class="item"
    >
      Four
    </span>
  </div>
</div>
`;

exports[`<BreadCrumb /> should allow to conditionnaly add items inside fragments 2`] = `
<div>
  <div
    class="junipero breadcrumb opened"
  >
    <span
      class="item"
    >
      One
    </span>
    <span
      class="item"
    >
      Two
    </span>
    <span
      class="item"
    >
      Three
    </span>
    <span
      class="item"
    >
      Four
    </span>
  </div>
</div>
`;

exports[`<BreadCrumb /> should render 1`] = `
<div>
  <div
    class="junipero breadcrumb"
  />
</div>
`;

exports[`<BreadCrumb /> should render a collapsed BreadCrumb 1`] = `
<div>
  <div
    class="junipero breadcrumb collapsed"
  >
    <span
      class="item"
    >
      One
    </span>
    <span
      class="item"
    >
      Two
    </span>
    <a
      class="item"
    >
      ...
    </a>
    <span
      class="item"
    >
      Four
    </span>
  </div>
</div>
`;

exports[`<BreadCrumb /> should render a collapsed BreadCrumb 2`] = `
<div>
  <div
    class="junipero breadcrumb opened"
  >
    <span
      class="item"
    >
      One
    </span>
    <span
      class="item"
    >
      Two
    </span>
    <span
      class="item"
    >
      Three
    </span>
    <span
      class="item"
    >
      Four
    </span>
  </div>
</div>
`;

exports[`<BreadCrumb /> should render a list of BreadCrumb items 1`] = `
<div>
  <div
    class="junipero breadcrumb"
  >
    <span
      class="item"
    >
      test
    </span>
    <span
      class="item"
    >
      secondTest
    </span>
  </div>
</div>
`;
